src/host/process_launcher.h

```diff
@@ -63,30 +237,49 @@ namespace asynchost
       }
       argv.push_back(nullptr);
 
+      close_ptr<ProcessReader> stdout_reader("stdout");
+      close_ptr<ProcessReader> stderr_reader("stderr");
+      close_ptr<ProcessWriter> stdin_writer(std::move(entry.input));
+
       auto handle = new uv_process_t;
       handle->data = this;
 
+      uv_stdio_container_t stdio[3];
+      stdio[0].flags = (uv_stdio_flags)(UV_CREATE_PIPE | UV_READABLE_PIPE);
+      stdio[0].data.stream = stdin_writer->stream();
+
+      stdio[1].flags = (uv_stdio_flags)(UV_CREATE_PIPE | UV_WRITABLE_PIPE);
+      stdio[1].data.stream = stdout_reader->stream();
+
+      stdio[2].flags = (uv_stdio_flags)(UV_CREATE_PIPE | UV_WRITABLE_PIPE);
+      stdio[2].data.stream = stderr_reader->stream();
+
       uv_process_options_t options = {};
       options.file = argv.at(0);
       options.args = const_cast<char**>(argv.data());
       options.exit_cb = ProcessLauncher::on_process_exit;
+      options.stdio = stdio;
+      options.stdio_count = 3;
 
       auto rc = uv_spawn(uv_default_loop(), handle, &options);
-
       if (rc != 0)
       {
         LOG_FAIL_FMT("Error starting host process: {}", uv_strerror(rc));
         return;
       }
 
-      LOG_DEBUG_FMT(
+      LOG_INFO_FMT(
```

Comment:
+1
I think we should do the same for the exit message:https://github.com/microsoft/CCF/blob/ed445bee49ff96fdf1d4107aff6ef5683308ad24/src/host/process_launcher.h#L302-L307
Ideally, `LOG_FAIL_FMT` if the exit code is not 0, and `LOG_INFO_FMT` otherwise. That would allow better filtering for subprocess failures.

Reply:
Done.